pkg/admission/validate_bucketclass.go

```diff
@@ -0,0 +1,207 @@
+package admission
+
+import (
+	"encoding/json"
+	"fmt"
+
+	nbv1 "github.com/noobaa/noobaa-operator/v5/pkg/apis/noobaa/v1alpha1"
+	"github.com/noobaa/noobaa-operator/v5/pkg/bucketclass"
+	"github.com/noobaa/noobaa-operator/v5/pkg/util"
+	"github.com/sirupsen/logrus"
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// BucketClassValidator is the context of a bucketclass validation
+type BucketClassValidator struct {
+	BucketClass *nbv1.BucketClass
+	Logger      *logrus.Entry
+	arRequest   *admissionv1.AdmissionReview
+	arResponse  *admissionv1.AdmissionReview
+}
+
+// NewBucketClassValidator initializes a BucketClassValidator to be used for loading and validating a bucketclass
+func NewBucketClassValidator(arRequest admissionv1.AdmissionReview) *BucketClassValidator {
+	bcv := &BucketClassValidator{
+		arRequest: &arRequest,
+		Logger:    logrus.WithField("admission bucketclass validation", arRequest.Request.Namespace),
+	}
+	return bcv
+}
+
+// ValidateBucketClass call appropriate validations based on the operation
+func (bcv *BucketClassValidator) ValidateBucketClass() admissionv1.AdmissionReview {
+	bcv.arResponse = &admissionv1.AdmissionReview{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "AdmissionReview",
+			APIVersion: "admission.k8s.io/v1",
+		},
+		Response: &admissionv1.AdmissionResponse{
+			UID:     bcv.arRequest.Request.UID,
+			Allowed: true,
+			Result: &metav1.Status{
+				Message: "allowed",
+			},
+		},
+	}
+
+	switch bcv.arRequest.Request.Operation {
+	case admissionv1.Create:
+		bcv.ValidateCreate()
+	case admissionv1.Delete:
+		bcv.ValidateDelete()
+	default:
+		bcv.Logger.Error("Failed to identify bucketclass operation type")
+	}
+	return *bcv.arResponse
+}
+
+// SetValidationResult responsible of assinging the return values of a validation into the response appropriate fields
+func (bcv *BucketClassValidator) SetValidationResult(isAllowed bool, message string) {
+	bcv.arResponse.Response.Allowed = isAllowed
+	bcv.arResponse.Response.Result.Message = message
+}
+
+// DeserializeBC extract the bucketclass object from the request
+func (bcv *BucketClassValidator) DeserializeBC(rawBS []byte) *nbv1.BucketClass {
+	BC := nbv1.BucketClass{}
+	if err := json.Unmarshal(rawBS, &BC); err != nil {
+		bcv.Logger.Error("error deserializing bucketclass")
+	}
+	return &BC
+}
+
+// ValidateCreate runs all the validations tests for CREATE operations
+func (bcv *BucketClassValidator) ValidateCreate() {
+	bcv.BucketClass = bcv.DeserializeBC(bcv.arRequest.Request.Object.Raw)
+
+	if ok, message := bcv.ValidateQuota(); !ok {
+		bcv.SetValidationResult(ok, message)
+		return
+	}
+	if bcv.BucketClass.Spec.NamespacePolicy != nil {
```

Comment:
2 things here:
1. namespace bucketclass can be of type cache and then you need to validate its tiering
2. you should also test that backingstores exist and ready